experimental/dds/tree/src/SharedTree.ts

```diff
@@ -1548,51 +1560,108 @@ export class SharedTree extends SharedObject<ISharedTreeEvents> implements NodeI
 	 *
 	 * @param content - op to apply locally.
 	 */
-	protected applyStashedOp(op: unknown): void {
+	protected applyStashedOp(op: unknown): StashedLocalOpMetadata {
+		// In some scenarios, edit ops need to have their edits transformed before application and resubmission. The transformation
+		// occurs in this method, and the result is passed to `resubmitCore` via the return value of this function.
 		const sharedTreeOp = op as SharedTreeOp | SharedTreeOp_0_0_2;
 		switch (sharedTreeOp.type) {
 			case SharedTreeOpType.Edit: {
+				let stashedEdit: Edit<ChangeInternal> | undefined;
 				switch (this.writeFormat) {
 					case WriteFormat.v0_0_2:
 						switch (sharedTreeOp.version) {
 							case WriteFormat.v0_0_2: {
-								const edit = this.parseSequencedEdit(sharedTreeOp);
-								this.applyEditLocally(edit, undefined);
+								stashedEdit = this.parseSequencedEdit(sharedTreeOp);
 								break;
 							}
 							case WriteFormat.v0_1_1:
-								// TODO:#74390: Implement
 								fail('Received stashed op 0.1.1 before upgrade');
 							default:
 								fail('Unknown version');
 						}
 						break;
 					case WriteFormat.v0_1_1:
 						switch (sharedTreeOp.version) {
-							case WriteFormat.v0_0_2:
-								// TODO:#74390: Implement
-								fail('v0.1.1 does not support stashed ops.');
-							case WriteFormat.v0_1_1:
-								// TODO:#74390: Implement
-								fail('v0.1.1 does not support stashed ops.');
+							case WriteFormat.v0_0_2: {
+								// Use the IDs from the stashed ops as overrides for the equivalent new ops
+								stashedEdit = convertEditIds(sharedTreeOp.edit, (id) => this.generateNodeId(id));
+								break;
+							}
+							case WriteFormat.v0_1_1: {
+								assert(this.stashedIdCompressor !== null, 'Stashed op applied after expected window');
+								// Allocate
+								this.stashedIdCompressor ??= IdCompressor.deserialize(
+									this.idCompressor.serialize(false),
+									stashedSessionId,
+									sharedTreeOp.idRange.attributionId
+								);
+								// Ack
+								this.stashedIdCompressor.finalizeCreationRange(sharedTreeOp.idRange);
+								const stashedIdContext = getNodeIdContext(this.stashedIdCompressor);
+								// Apply
+								const normalizer: NodeIdNormalizer<OpSpaceNodeId> = {
+									localSessionId: this.idCompressor.localSessionId,
+									normalizeToSessionSpace: (id, _sessionId) => {
```

Comment:
It feels like we must have done this before, could we have a transferNormalizer creation method?

Reply:
Have we? This one:
1. Delegates the normalization to another normalizer, but is also generating IDs with overrides
2. Uses the local session ID from the original normalizer
3. And uses a normalizing session ID from the incoming op

So it's combining state from three different contexts and doing more than strictly normalizing. We'd have to write a function where all three of those are passed in, _and_ the normalizing function. I think at that point the overhead of understanding/documenting the function might exceed the value of deduplication.

Reply:
Resolving, but feel free to re-open if you disagree.